From 176fc243f83c628c6da55c217a785aa1fd15a262 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <nathan@kernel.org>
Date: Fri, 6 May 2022 08:03:58 -0700
Subject: [PATCH 6/7] nvme: Ensure ret is always initialized in
 nvme_ns_head_chr_uring_cmd()

Clang warns:

  drivers/nvme/host/ioctl.c:674:6: error: variable 'ret' is used uninitialized whenever 'if' condition is false [-Werror,-Wsometimes-uninitialized]
          if (ns)
              ^~
  drivers/nvme/host/ioctl.c:677:9: note: uninitialized use occurs here
          return ret;
                 ^~~
  drivers/nvme/host/ioctl.c:674:2: note: remove the 'if' if its condition is always true
          if (ns)
          ^~~~~~~
  drivers/nvme/host/ioctl.c:672:9: note: initialize the variable 'ret' to silence this warning
          int ret;
                 ^
                  = 0
  1 error generated.

Initialize ret to zero so that it cannot be used uninitialized.

Fixes: 271ee3df5cca ("nvme: wire-up uring-cmd support for io-passthru on char-device.")
Link: https://github.com/ClangBuiltLinux/linux/issues/1630
Reported-by: kernel test robot <lkp@intel.com>
Signed-off-by: Nathan Chancellor <nathan@kernel.org>
Link: https://lore.kernel.org/r/20220506150357.2443040-1-nathan@kernel.org
---
 drivers/nvme/host/ioctl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/nvme/host/ioctl.c b/drivers/nvme/host/ioctl.c
index 4b91e452768b..c118559d47f0 100644
--- a/drivers/nvme/host/ioctl.c
+++ b/drivers/nvme/host/ioctl.c
@@ -688,7 +688,7 @@ int nvme_ns_head_chr_uring_cmd(struct io_uring_cmd *ioucmd,
 	struct nvme_ns_head *head = container_of(cdev, struct nvme_ns_head, cdev);
 	int srcu_idx = srcu_read_lock(&head->srcu);
 	struct nvme_ns *ns = nvme_find_path(head);
-	int ret;
+	int ret = 0;
 
 	if (ns)
 		ret = nvme_ns_uring_cmd(ns, ioucmd, issue_flags);
-- 
2.36.0

